### PR TITLE
Assignment 2

### PR DESCRIPTION
**Pull request by RaulPetr**

### CODE REVIEW

File: Assignment 2/Petrovan Raul/backend/src/main/java/com/lab4/demo/report/ReportController.java [hunk 1]
```diff
@@ -0,0 +1,62 @@
...
+    private ReportServiceFactory reportServiceFactory;
+    @Autowired
+    private BookService bookService;
+
+    @GetMapping(PDF)
+    public ResponseEntity<Resource> getReportPDF() throws IOException {
+        System.out.println("PDF");
+        bookService.export(ReportType.PDF);
+
+        File file = new File(REPORT_LOCAL_PATH + REPORT_LOCAL_PDF);
```
Comment (Alex-Cosma) [inline Assignment 2/Petrovan Raul/backend/src/main/java/com/lab4/demo/report/ReportController.java:41]: Oh, you did this ... 
Well, this is quite bad, here's why:
- this functionality should be linked directly with the "export" one from `BookController`
- in what you did, you create the same file all the time and overwrite it just so that you can download it
- furthermore, you are calling the `bookService.export(...)` function again, even though, it is called in the `EXPORT_REPORT` endpoint as well

Long story short, you should not have two (or 3, in this case) endpoints for exporting the report, just one.
You should also combine PDF with CSV as the code is mostly the same, the only substantial difference being the `contentType` which can easy be `if-ed`

**Review by Alex-Cosma — COMMENTED** [review]
Godo work, it'll be even better next time
